File: CinemaTicket/src/app/TicketPriceCalculationService.ts
import Calculator from '../domain/models/calculator';
import MovieRepository from '../domain/repositories/movieRepository';
import PriceTable from '../domain/models/priceTable';
import { Visitor } from '../domain/models/visitor';
import Discount from '../domain/models/discount';

export default class TicketPriceCalculationService {
  movieRepository: MovieRepository

  constructor(movieRepository: MovieRepository) {
    this.movieRepository = movieRepository;
  }

  calculate(visitor: Visitor, discount: Discount, movieId: number, screeningTime: Date): number {
    const movie = this.movieRepository.get(movieId);
    const priceList = PriceTable.get(screeningTime);
    const price = Calculator.calc(movie, visitor, priceList, discount);
    return price;
  }
}

File: CinemaTicket.kt/src/main/kotlin/com/example/cinematicket/app/TicketPriceCalculationService.kt
package app

import domain.models.*
import java.time.LocalDateTime

class TicketPriceCalculationService(val movieRepository: MovieRepository) {
    fun calculate(visitor: Visitor, discount: Discount, movieId: Int, screeningTime: LocalDateTime) :Int {
        val movie = movieRepository.get(movieId)
        val priceTable = PriceTable().get(screeningTime)
        val price = calc(movie, visitor, priceTable, discount)
        return price
    }
}
